python/ray/_private/runtime_env/nsight.py

```diff
@@ -0,0 +1,132 @@
+import os
+import logging
+import subprocess
+import copy
+from typing import Tuple, List, Dict, Optional
+
+from ray._private.runtime_env.context import RuntimeEnvContext
+from ray._private.runtime_env.plugin import RuntimeEnvPlugin
+from ray._private.utils import (
+    try_to_create_directory,
+)
+
+default_logger = logging.getLogger(__name__)
+
+NSIGHT_DEFAULT_CONFIG = {
+    "-t": "cuda,cudnn,cublas,nvtx",
+    "-o": "'worker_process_%p'",
+    "--cudabacktrace": "all",
+    "--stop-on-exit": "true",
+}
+
+
+def check_nsight_script(nsight_config: Dict[str, str]) -> Tuple[bool, str]:
+    """
+    Function to check if the provided nsight options are
+    valid nsys profile options and if nsys profile is installed
+
+    The function returns:
+    - bool: True if the nsight_config is valid option
+    - str: error message if the nsight_config is invalid
+    """
+
+    # use empty as nsight report test filename
+    nsight_config_copy = copy.deepcopy(nsight_config)
+    nsight_config_copy["-o"] = "empty"
+    nsight_cmd = parse_nsight_config(nsight_config_copy)
+    try:
+        nsight_cmd = nsight_cmd + ["python", "-c", '""']
+        result = subprocess.run(
+            nsight_cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+        if result.returncode == 0:
+            subprocess.run(["rm", "empty.nsight-rep"])
```

Comment:
can you create a file inside profile folder?

Reply:
I think there are potential risk of having leaked files, and we can handle it easily with our GC reconciler when we have one

Reply:
Sounds good